OpenRA.Mods.Common/Traits/Buildings/BaseProvider.cs

```diff
@@ -30,22 +30,26 @@ public class BaseProviderInfo : ITraitInfo
 	public class BaseProvider : ITick, INotifyCreated, IRenderAboveShroudWhenSelected, ISelectionBar
 	{
 		public readonly BaseProviderInfo Info;
+		readonly MapBuildRadius mapBuildRadius;
 		readonly DeveloperMode devMode;
 		readonly Actor self;
+		readonly bool allyBuildEnabled;
+		readonly bool buildRadiusEnabled;
 
 		Building building;
 
 		int total;
 		int progress;
-		bool allyBuildEnabled;
 
 		public BaseProvider(Actor self, BaseProviderInfo info)
 		{
 			Info = info;
 			this.self = self;
 			devMode = self.Owner.PlayerActor.Trait<DeveloperMode>();
 			progress = total = info.InitialDelay;
-			allyBuildEnabled = self.World.WorldActor.Trait<MapBuildRadius>().AllyBuildRadiusEnabled;
+			mapBuildRadius = self.World.WorldActor.Trait<MapBuildRadius>();
```

Comment:
Actually, we only use this in the constructor. So I'd suggest you make this a local `var` and drop the `readonly MapBuildRadius mapBuildRadius;`.